share/getters/shrex.go

```diff
@@ -34,6 +34,11 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
 	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
 		peer, setStatus, err := sg.peerManager.Peer(ctx, root.Hash())
```

Comment:
AFAICS, Peer method handles context correctly, and usually, such ctx check means the call down the stack is blocking. Why do we need to do it?

Reply:
We need it in the case that there is a peer in the pool. `Peer` returns a peer before it checks the context in this case.